docs/stacks-guide-aws.md

```diff
@@ -0,0 +1,896 @@
+AWS_CREDS_BASE64=$(cat ${HOME}/.aws/credentials | awk '/["$aws_profile"]/ {getline; print $0}' | awk 'NR==1{print "[default]"}1' | base64 | tr -d "\n")
+
+# retrieve the profile's region from config
+AWS_REGION=$(awk '/["$aws_profile"]/ {getline; print $3}' ${HOME}/.aws/config)
+```
+
+At this point, the region and the encoded credentials are stored in respective
+environment variables. Next, we'll need to create an instance of [`aw
+provider`]:
+
+```bash
+cat > ${WORK_DIR}/provider.yaml <<EOF
+---
+apiVersion: v1
+data:
+  credentials: ${AWS_CREDS_BASE64}
+kind: Secret
+metadata:
+  name: aws-user-creds
+  namespace: ${INFRA_NAMESPACE}
+type: Opaque
+---
+apiVersion: aws.crossplane.io/v1alpha2
+kind: Provider
+metadata:
+  name: aws-provider
+  namespace: ${INFRA_NAMESPACE}
+spec:
+  credentialsSecretRef:
+    key: credentials
+    name: aws-user-creds
+  region: ${AWS_REGION}
+EOF
+
+# apply it to the cluster:
+kubectl apply -f "${WORK_DIR}/provider.yaml"
+```
+
+<details> <summary>output</summary>
+
+```bash
+secret/aws-user-creds created
+provider.aws.crossplane.io/aws-provider created
+```
+
+</details>
+
+## Set Up Network Configuration
+
+When configured in AWS, WordPress resources map to an EKS cluster and an RDS
+database. In order to make the RDS instance accessible by  EKS cluster, they
+both need to live within the same VPC. However VPC is not the only AWS resource
+that needs to be created to enable inter-resource connecitivity. In general, a
+*Network Configuration*, which is built of a set of VPCs, Subnets, Security
+Groups, Route Tables, IAM Roles and other resources, is requited for this
+purpose. For more information, see [AWS resource connectivity] design document.
+
+In this section we will build a simple network configuration, by creating AWS
+resources that are managed by Crossplane. There are a couple of challenges when
+creating these resources:
+
+- There is a dependency between some of these resources. For instance, a Subnet
+  is dependent on a VPC, so creating a Subnet needs to be done after creating
+  the VPC.
+
+  To solve this issue, we will need to create the resoruces in order, so the
+  depependent resources are provisioined after creating their dependencies.
+  Since provisioning a resource might take some time, we need to make sure the
+  resource is ready, before moving forward to the next step. Let's create the
+  following function for this purpose:
+
+  ```bash
+  # apply_and_wait_until_ready accepts the yaml file name as an argument
+  # and then applies the yaml object. Then waits until the resource status 
+  # becomes Ready
+  function apply_and_wait_until_ready {
+    kubectl apply -f "$1"
+    kubectl wait --for=condition=Ready -f "$1"
+  }
+  ```
+
+- Some of these resources have identifying attributes that are
+  non-deterministic. In other words, they become known after the resource is
+  provisioined. For instance, a VPCID which is consumed by all the dependent
```

Comment:
I like to avoid using the word `dependent` because it is so close to `dependency`, so I would probably word `a VPCID which is consumed by all the dependent resources` more like `a VPCID which is consumed by many resources`